File: WallAI.HostedSimulation.Web/app/Rpc/RpcResponseEvent.ts
import { RpcResponse } from "./RpcResponse";
import { RpcResponseType } from "./RpcResponseTypes";

export class RpcResponseEvent<T> extends RpcResponse {
    data: T;

    constructor(id: string, data: T) {
        super(id, RpcResponseType.Event)
        this.data = data;
    }
}

File: WallAI.HostedSimulation/Servers/Rpc/RpcResponseEvent.cs
using System;
using System.Runtime.Serialization;

namespace WallAI.HostedSimulation.Servers.Rpc
{
    [DataContract]
    public class RpcResponseEvent<T> : RpcResponse
    {
        [DataMember(Name = "data")]
        public readonly T Data;

        public RpcResponseEvent(Guid id, T data) : base(id, RpcResponseType.Event)
        {
            Data = data;
        }
    }
}